examples/protocol_specification_ex/output/two_party_negotiation/message.py

```diff
@@ -43,47 +43,61 @@ def __init__(self, message_id: int, target: int, performative: str, contents: Di
         assert self.check_consistency()
 
     @property
-    def performatives(self) -> set:
+    def message_id(self) -> int:
+        """Get the message_id of the message."""
+        assert self.is_set("message_id"), "message_id is not set"
+        return cast(int, self.get("message_id"))
+
+    @property
+    def target(self) -> int:
+        """Get the target of the message."""
+        assert self.is_set("target"), "target is not set."
+        return cast(int, self.get("target"))
+
+    @property
+    def performative(self) -> str:
+        """Get the performative of the message."""
+        assert self.is_set("performative"), "performative is not set"
+        return cast(str, self.get("performative"))
+
+    @property
+    def contents(self) -> Dict:
+        """Get the contents of the message."""
+        assert self.is_set("contents"), "contents is not set"
+        return cast(Dict, self.get("contents"))
+
+    @property
+    def performatives_definition(self) -> set:
```

Comment:
Why not call it `permitted_performatives`? Also, if `permitted_performatives` is only used in the message class I would make it 'private': `_permitted_performatives` and not a property

Reply:
At some point I decided to differentiate the fields of a message being validated vs the correct definitions by appending the laters' variable names with "_definition", e.g. "contents": the contents of the message, and "contents_definition": its definition which comes from the specification file.

Happy to change for all instances to "valid_..." or "allowed_...", "permitted_...", "defined_...".

Reply:
I think the `definition` is a bit ambiguous. Naming-wise I prefer `allowed_` or `permitted_` as its clearer. However, we can further improve this by making them class rather than instance variables. The `permitted_variables` is a class variable as it applies to all instances of `SomeProtocolMessage`.

Reply:
ok I change to 'allowed_X'.

I will make it a class property.

Reply:
Done